src/agent/src/health_management.c

```diff
@@ -469,42 +476,52 @@ static bool CheckShellBinary()
 {
     bool result = false;
 
-    const char* path = ADUSHELL_FILE_PATH;
+    const ADUC_ConfigInfo* config = ADUC_ConfigInfo_GetInstance();
 
-    if (SystemUtils_IsFile(path, NULL))
+    if (config == NULL)
     {
-        if (!PermissionUtils_CheckOwnerUid(path, 0 /* root */))
+        Log_Error("ADUC_ConfigInfo singleton hasn't been initialized.");
+        goto done;
+    }
+
+    if (SystemUtils_IsFile(config->aduShellFilePath, NULL))
+    {
+        if (!PermissionUtils_CheckOwnerUid(config->aduShellFilePath, 0 /* root */))
         {
-            Log_Error("'%s' has incorrect UID.", path);
+            Log_Error("'%s' has incorrect UID.", config->aduShellFilePath);
             goto done;
         }
 
-        if (!PermissionUtils_CheckOwnership(path, NULL /* user */, ADUC_FILE_GROUP))
+        if (!PermissionUtils_CheckOwnership(config->aduShellFilePath, NULL /* user */, ADUC_FILE_GROUP))
         {
-            Log_Error("'%s' has incorrect group owner.", path);
+            Log_Error("'%s' has incorrect group owner.", config->aduShellFilePath);
             goto done;
         }
 
         // Needs set-uid, user read, and group read + execute.
         // Note: other has no permission bits set.
         const mode_t expected_permissions = S_ISUID | S_IRUSR | S_IXUSR | S_IRGRP | S_IXGRP;
 
-        if (!PermissionUtils_VerifyFilemodeExact(path, expected_permissions))
+        if (!PermissionUtils_VerifyFilemodeExact(config->aduShellFilePath, expected_permissions))
         {
-            Log_Error("Lookup failed or '%s' has incorrect permissions (expected: 0%o)", path, expected_permissions);
+            Log_Error(
+                "Lookup failed or '%s' has incorrect permissions (expected: 0%o)",
+                config->aduShellFilePath,
```

Comment:
why customer needs to override aduShell file path? Might be better to invest in installing all ADU files under different prefix e.g. /opt/my/path/{var/lib/adu/, etc/adu/, usr/bin/}